File: Dec2020/26Dec/GoalParserInterpretation.java
/*
Problem Name: 1678. Goal Parser Interpretation
Problem Link: https://leetcode.com/problems/goal-parser-interpretation/
*/
class Solution {
    public String interpret(String command) {
        String ans = "";
        char arr[] = command.toCharArray();
        for(int i=0;i<arr.length;){
            if(arr[i] == 'G'){
                ans += "G";
                i++;
            }
            else if(arr[i] == '(' && arr[i+1] == ')'){
                ans += "o";
                i += 2;
            }
            else{
                ans += "al";
                i += 4;
            }
        }
        return ans;
    }
}

File: Dec2020/26Dec/GoalParserInterpretation.cpp
/*
Problem name: 1678. Goal Parser Interpretation
Problem link: https://leetcode.com/problems/goal-parser-interpretation/
*/

class Solution {
public:
    string interpret(string command) {
        string ans="";
        for(int i=0;i<command.size();++i){
            if(command[i] == 'G')
                ans += "G";
            else if(command[i] == '(' && command[i+1] == ')' ){
                ans += "o";
                i += 1;
            }
            else{
                ans += "al";
                i += 3;
            }
        }
        return ans;
    }
};
